<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>364</width>
    <height>263</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QStackedWidget" name="stackedWidget">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>361</width>
      <height>231</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="page">
     <widget class="QLineEdit" name="mysql_port_edit">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>60</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="sftp_port_edit">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>90</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>60</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>MySQL Port:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>90</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>SFTP Port:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="initiate_server">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>190</y>
        <width>121</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Initiate Server</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Server Initiate</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page_2">
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Server Status</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>130</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Port Number</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>80</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Port Number</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="sftp_port_label">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>130</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLabel" name="mysql_port_label">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>80</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>50</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>MySQL:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>100</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>SFTP:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="user_button">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>180</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>User</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_2">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>180</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Server Restart</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="user_page">
     <widget class="QListView" name="user_list_view">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>131</width>
        <height>181</height>
       </rect>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
     </widget>
     <widget class="QPushButton" name="user_save_button">
      <property name="geometry">
       <rect>
        <x>234</x>
        <y>200</y>
        <width>121</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Save and Restart</string>
      </property>
     </widget>
     <widget class="QPushButton" name="add_user_button">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>200</y>
        <width>21</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>+</string>
      </property>
     </widget>
     <widget class="QPushButton" name="delete_user_button">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>200</y>
        <width>21</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>-</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_6">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>200</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Cancel</string>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>20</y>
        <width>56</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>ID</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="user_id_edit">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>20</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="user_confirm_button">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>150</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>confirm</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="radio_visitor">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>130</y>
        <width>90</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Visitor</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>70</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>PW confirm</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="new_password_edit">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>50</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
      <property name="inputMethodHints">
       <set>Qt::ImhHiddenText|Qt::ImhNoAutoUppercase|Qt::ImhNoPredictiveText|Qt::ImhSensitiveData</set>
      </property>
      <property name="echoMode">
       <enum>QLineEdit::Password</enum>
      </property>
     </widget>
     <widget class="QRadioButton" name="radio_researcher">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>110</y>
        <width>90</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Researcher</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="radio_manager">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>90</y>
        <width>90</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Manager</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="password_confirm_edit">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>70</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
      <property name="inputMethodHints">
       <set>Qt::ImhHiddenText|Qt::ImhNoAutoUppercase|Qt::ImhNoPredictiveText|Qt::ImhSensitiveData</set>
      </property>
      <property name="echoMode">
       <enum>QLineEdit::Password</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>50</y>
        <width>56</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>New PW</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>364</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>user_id_edit</tabstop>
  <tabstop>new_password_edit</tabstop>
  <tabstop>password_confirm_edit</tabstop>
  <tabstop>radio_manager</tabstop>
  <tabstop>radio_researcher</tabstop>
  <tabstop>radio_visitor</tabstop>
  <tabstop>user_confirm_button</tabstop>
  <tabstop>pushButton_6</tabstop>
  <tabstop>user_save_button</tabstop>
  <tabstop>delete_user_button</tabstop>
  <tabstop>add_user_button</tabstop>
  <tabstop>mysql_port_edit</tabstop>
  <tabstop>user_list_view</tabstop>
  <tabstop>pushButton_2</tabstop>
  <tabstop>sftp_port_edit</tabstop>
  <tabstop>user_button</tabstop>
  <tabstop>initiate_server</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>initiate_server</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>server_initiation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>273</x>
     <y>225</y>
    </hint>
    <hint type="destinationlabel">
     <x>250</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>user_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>user_button_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>279</x>
     <y>225</y>
    </hint>
    <hint type="destinationlabel">
     <x>274</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>delete_user_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>delete_user_button_click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>119</x>
     <y>228</y>
    </hint>
    <hint type="destinationlabel">
     <x>120</x>
     <y>285</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>add_user_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>add_user_button_click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>144</x>
     <y>238</y>
    </hint>
    <hint type="destinationlabel">
     <x>144</x>
     <y>331</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>user_list_view</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>MainWindow</receiver>
   <slot>user_list_view_double_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>129</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>user_confirm_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>user_confirm_button_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>330</x>
     <y>193</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>-140</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>user_save_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>user_save_button_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>304</x>
     <y>232</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>server_restart_button_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>157</x>
     <y>222</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>user_cancel_button_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>232</y>
    </hint>
    <hint type="destinationlabel">
     <x>201</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>server_initiation()</slot>
  <slot>user_button_clicked()</slot>
  <slot>delete_user_button_click()</slot>
  <slot>add_user_button_click()</slot>
  <slot>user_list_view_double_clicked()</slot>
  <slot>user_confirm_button_clicked()</slot>
  <slot>user_save_button_clicked()</slot>
  <slot>server_restart_button_clicked()</slot>
  <slot>user_cancel_button_clicked()</slot>
 </slots>
</ui>
